mirny.py

```diff
@@ -310,23 +310,59 @@ def __init__(self, platform):
         clk = platform.request("clk")
         clk_div = TSTriple()
         self.specials += clk_div.get_tristate(clk.div)
-        # in_sel: 00: XO, 01: n/a (SMA+XO), 10: MMXC, 11: SMA
+        # in_sel: 00: XO, 01: n/a (SMA+XO), 10: MMCX, 11: SMA
         # dividers: 00(z): 1, 01(z): 1, 10(low): 2, 11(high) 4
         self.comb += [
             Cat(clk.in_sel, clk_div.o, clk_div.oe).eq(regs[1].write[4:8]),
             platform.request("fsen").eq(~regs[1].write[9]),
         ]
 
-        for i, m in enumerate(platform.request("mezz_io")):
-            tsi = TSTriple()
-            self.specials += tsi.get_tristate(m)
-            self.comb += [
-                tsi.o.eq(regs[3].write[i] | (0 if i >= 4 else
-                    (regs[1].write[11] & eem[i + 4].i))),
-                regs[3].read[i].eq(tsi.i),
-                tsi.oe.eq(regs[3].write[i + 8]),
-                regs[3].read[i + 8].eq(tsi.oe),
-            ]
+        if legacy_almazny:
+            # 6 signals
+            for i, m in enumerate(platform.request("almazny_io")):
```

Comment:
Name them explicitly for consistency with ``noe`` and ``nsrclr``, don't just group them into one ``io`` vector.
You can still make a list of them and then use enumerate.